doc: add Windows-specific info to `subprocess.kill` (#34867)
nodejs/node: doc/api/child_process.md

See kill(2) for reference.

> On Windows the child process will be forcefully terminated since

Reviewer: ```suggestion
On Windows, the child process is forcefully terminated because
```

> `SIGKILL` and `SIGTERM` signals are not available.

Reviewer 2: Should this say `SIGINT` and `SIGTERM` instead? 
If I'm not mistaken, `SIGINT` isn't available here either.
And `SIGKILL`, if available, forcibly terminates (so it's unavailability doesn't help explain why child process must be forcefully terminated.. `SIGKILL` wouldn't be useful for avoiding forceful termination anyways).

See kill(2) for reference.

> On Windows, the child process is forcefully terminated because
> `SIGKILL` and `SIGTERM` signals are not available.

Reviewer 2: @joaolucasl I was thinking something like this:

```suggestion
On Windows, where POSIX signals do not exist, the `signal` argument will be
ignored, and the process will be killed forcefully and abruptly (similar to
`'SIGKILL'`).
```

Author: I Iike that. Might work!

[`process.env`]: process.html#process_process_env
[`process.execPath`]: process.html#process_process_execpath
[`process.send()`]: process.html#process_process_send_message_sendhandle_options_callback

> [Signal Events]: process.html#process_signal_events

Reviewer: Once you've rebased, be sure to run `make lint-md` or `make test-doc` because I'm pretty sure the linter is going to say this line needs to be moved above all the others (because the are in ASCII order and `S` comes before ``` ` ```).
